### PR TITLE
Wrong exports assignment when reexporting across siblings 

### PR DESCRIPTION
**Pull request by mischnic**
# ↪️ Pull Request

1. When a bundle exports reexports some import that is in turn in a sibling, this happened:

```js
import { $id$init } from "./xyz.29adda.js";

$id2$exports.foo = /* $id2$exports$foo = */$id$init().foo;

export function $id2$init() { 
	return $id2$exports
};
```

The commented out part is now omitted if there is no such binding (as is the case in this situation). This would only throw on strict mode (= esmodule).
I've changed the test-utils to always run scripts in strict mode, so that we can catch this (and also other things in the future).

2. In the test case for 1), the siblings were inserted in the wrong order. I've moved the sorting in `getBundlesInBundleGrouop` from the public BundleGraph into the internalBundleGraph so that `getSiblingBundles` respects it as well.
I'm not entirely sure why I needed to add a `reverse()` to the `getSiblingBundles` call in `findReachableBundleWithAsset`.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.96s | -129.00ms |
| Cached | 2.57s | +118.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 299.00ms | +42.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 298.00ms | +40.00ms ⚠️ |
| dist/legacy/index.42516b4a.js | 1.10kb | +0.00b | 1.71s | -123.00ms 🚀 |
| dist/modern/index.8a1ced31.js | 1.10kb | +0.00b | 1.71s | -124.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 867.00ms | +139.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 864.00ms | +138.00ms ⚠️ |
| dist/legacy/index.0e158b6b.css | 77.00b | +0.00b | 1.60s | -228.00ms 🚀 |
| dist/modern/index.e480862d.css | 77.00b | +0.00b | 1.59s | +89.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 269.00ms | +65.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 269.00ms | +65.00ms ⚠️ |
| dist/legacy/index.42516b4a.js | 1.10kb | +0.00b | 462.00ms | +98.00ms ⚠️ |
| dist/modern/index.8a1ced31.js | 1.10kb | +0.00b | 461.00ms | +98.00ms ⚠️ |
| dist/legacy/index.html | 701.00b | +0.00b | 464.00ms | +94.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 461.00ms | +98.00ms ⚠️ |
| dist/legacy/index.0e158b6b.css | 77.00b | +0.00b | 462.00ms | +94.00ms ⚠️ |
| dist/modern/index.e480862d.css | 77.00b | +0.00b | 461.00ms | +98.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 32.85s | +1.18s |
| Cached | 2.74s | +15.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/PermalinkedComment.c85d137d.js | 4.22kb | +0.00b | 12.74s | +6.49s ⚠️ |
| dist/UserProfile.30e15baf.js | 1.62kb | +0.00b | 12.74s | +6.49s ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 165.00ms | -20.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/PermalinkedComment.c85d137d.js | 4.22kb | +0.00b | 374.00ms | +35.00ms ⚠️ |
| dist/UserProfile.30e15baf.js | 1.62kb | +0.00b | 374.00ms | +35.00ms ⚠️ |
| dist/NotFound.5aa6b041.js | 387.00b | +0.00b | 374.00ms | +35.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 219.00ms | -21.00ms 🚀 |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 3.29m | -2.49s |
| Cached | 5.60s | +160.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.3417749f.js | 2.73mb | -42.00b 🚀 | 1.86m | -1.10s |
| dist/pdfRenderer.cd742f5e.js | 1.11mb | +0.00b | 1.04m | -4.36s 🚀 |
| dist/editorView.ae59b49f.js | 585.98kb | +0.00b | 1.53m | -6.16s 🚀 |
| dist/popup.4a66e15e.js | 170.62kb | +0.00b | 1.04m | +21.80s ⚠️ |
| dist/Toolbar.33a83ae3.js | 92.34kb | +0.00b | 1.88m | +41.95s ⚠️ |
| dist/card.1007a95f.js | 52.86kb | +0.00b | 1.07m | -5.33s 🚀 |
| dist/DatePicker.cbe44a9b.js | 21.17kb | +0.00b | 13.14s | -787.00ms 🚀 |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 13.14s | -788.00ms 🚀 |
| dist/dropzone.c8c03b6a.js | 14.36kb | +0.00b | 1.07m | -4.84s 🚀 |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 1.08m | -48.22s 🚀 |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 17.79s | -47.07s 🚀 |
| dist/component.11476a92.js | 6.05kb | +0.00b | 8.28s | +662.00ms ⚠️ |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 20.11s | +1.49s ⚠️ |
| dist/EmojiPickerComponent.624bf3f4.js | 3.58kb | +0.00b | 13.14s | -788.00ms 🚀 |
| dist/index.393afe59.css | 3.46kb | +0.00b | 8.14s | -1.26m 🚀 |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 1.03m | +24.63s ⚠️ |
| dist/clipboard.0244488f.js | 2.75kb | +0.00b | 1.05m | -4.54s 🚀 |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 8.21s | +4.53s ⚠️ |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 35.77s | +15.24s ⚠️ |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 8.16s | +787.00ms ⚠️ |
| dist/date.d49e64f8.js | 1.73kb | +0.00b | 8.96s | +557.00ms ⚠️ |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 8.28s | -1.09s 🚀 |
| dist/browser.16c438d4.js | 1.61kb | +0.00b | 1.05m | -4.48s 🚀 |
| dist/workerHasher.f66f00c7.js | 1.54kb | +0.00b | 1.07m | -48.41s 🚀 |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 10.09s | +2.11s ⚠️ |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 8.28s | -1.09s 🚀 |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 10.09s | +2.11s ⚠️ |
| dist/16.ab5c0088.js | 1.27kb | +0.00b | 8.79s | +567.00ms ⚠️ |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 13.14s | +4.08s ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 8.56s | +576.00ms ⚠️ |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 8.28s | -1.44s 🚀 |
| dist/16.1b342686.js | 1.13kb | +0.00b | 10.09s | +2.11s ⚠️ |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 10.11s | +2.13s ⚠️ |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 8.28s | -1.09s 🚀 |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 3.79s | -3.59s 🚀 |
| dist/component.40194190.js | 1.07kb | +0.00b | 8.16s | +787.00ms ⚠️ |
| dist/layout.ac6734f9.js | 1.05kb | +0.00b | 9.08s | +555.00ms ⚠️ |
| dist/panel-error.c4b05f6f.js | 906.00b | +0.00b | 9.19s | +502.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.553055f6.js | 837.00b | +0.00b | 1.05m | +22.99s ⚠️ |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 18.74s | +1.31s ⚠️ |
| dist/media-card-analytics-error-boundary.57503222.js | 833.00b | +0.00b | 1.07m | -48.41s 🚀 |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 1.07m | +50.29s ⚠️ |
| dist/simpleHasher.32d9352e.js | 528.00b | +0.00b | 1.07m | -48.40s 🚀 |
| dist/index.html | 119.00b | +0.00b | 2.14s | -1.91s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.3417749f.js | 2.73mb | -42.00b 🚀 | 712.00ms | +96.00ms ⚠️ |
| dist/pdfRenderer.cd742f5e.js | 1.11mb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/editorView.ae59b49f.js | 585.98kb | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/popup.4a66e15e.js | 170.62kb | +0.00b | 709.00ms | +97.00ms ⚠️ |
| dist/EmojiPickerComponent.bab709db.js | 97.89kb | +0.00b | 711.00ms | +98.00ms ⚠️ |
| dist/Toolbar.33a83ae3.js | 92.34kb | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/media-viewer.1d9322e7.js | 69.09kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/card.ce67dc35.js | 55.09kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/card.1007a95f.js | 52.86kb | +0.00b | 709.00ms | +104.00ms ⚠️ |
| dist/Modal.e3bd77d0.js | 41.40kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/component.59edff1f.js | 30.78kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/component.37867f50.js | 22.51kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/DatePicker.cbe44a9b.js | 21.17kb | +0.00b | 711.00ms | +98.00ms ⚠️ |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 711.00ms | +98.00ms ⚠️ |
| dist/js.48be70a4.js | 16.40kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/dropzone.c8c03b6a.js | 14.36kb | +0.00b | 709.00ms | +104.00ms ⚠️ |
| dist/ui.cee430a3.js | 13.97kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.9af6e5fc.js | 8.67kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/component.11476a92.js | 6.05kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/media-viewer.bc661bce.js | 3.92kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/EmojiPickerComponent.624bf3f4.js | 3.58kb | +0.00b | 711.00ms | +98.00ms ⚠️ |
| dist/index.393afe59.css | 3.46kb | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/png-chunks-extract.8d15a914.js | 3.41kb | +0.00b | 711.00ms | +96.00ms ⚠️ |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 709.00ms | +97.00ms ⚠️ |
| dist/Modal.2835744c.js | 3.08kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/clipboard.0244488f.js | 2.75kb | +0.00b | 709.00ms | +97.00ms ⚠️ |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/EmojiUploadComponent.88a444df.js | 2.18kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 709.00ms | +96.00ms ⚠️ |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/date.d49e64f8.js | 1.73kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.f40c1329.js | 1.73kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/images.80270b1f.js | 1.67kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/feedback.8b7c7928.js | 1.64kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/browser.16c438d4.js | 1.61kb | +0.00b | 709.00ms | +97.00ms ⚠️ |
| dist/workerHasher.9bb425b1.js | 1.54kb | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/workerHasher.f66f00c7.js | 1.54kb | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/list-number.4632b459.js | 1.46kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/status.cccf04be.js | 1.46kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/code.af9ca2b3.js | 1.38kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.a4b392c2.js | 1.32kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.249c644a.js | 1.31kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/link.d12cc130.js | 1.30kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/heading6.4c4017db.js | 1.30kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/heading3.93606b9c.js | 1.29kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/16.ab5c0088.js | 1.27kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/emoji.e603e70e.js | 1.23kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/16.afee1494.js | 1.23kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.e60b4d77.js | 1.21kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.7c7a40c9.js | 1.19kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/16.06d5a67b.js | 1.18kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/heading5.dde15cec.js | 1.17kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.6a25b970.js | 1.15kb | +0.00b | 709.00ms | +96.00ms ⚠️ |
| dist/16.2096b9f3.js | 1.14kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.1b342686.js | 1.13kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/heading2.e219309a.js | 1.11kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/component.40194190.js | 1.07kb | +0.00b | 712.00ms | +97.00ms ⚠️ |
| dist/mention.b1828447.js | 1.07kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/heading4.5316f6f2.js | 1.06kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/16.317b9164.js | 1.05kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/layout.ac6734f9.js | 1.05kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/divider.d2be55a1.js | 1.03kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/quote.f4e09143.js | 1.02kb | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/action.cb287eb0.js | 1.00kb | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/decision.91f3af14.js | 1010.00b | +0.00b | 712.00ms | +98.00ms ⚠️ |
| dist/panel-warning.68b04096.js | 1008.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/list.057cb664.js | 978.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/heading1.fa66b375.js | 976.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/panel-error.c4b05f6f.js | 906.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/panel.9dbc048e.js | 897.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/table.b35545be.js | 889.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/panel-success.7eb0809a.js | 845.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/panel-note.ec8386b7.js | 841.00b | +0.00b | 711.00ms | +97.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.553055f6.js | 837.00b | +0.00b | 709.00ms | +97.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.57503222.js | 833.00b | +0.00b | 709.00ms | +104.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.bbe40a02.js | 757.00b | +0.00b | 709.00ms | +96.00ms ⚠️ |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 711.00ms | +98.00ms ⚠️ |
| dist/simpleHasher.f8e69464.js | 528.00b | +0.00b | 710.00ms | +97.00ms ⚠️ |
| dist/simpleHasher.32d9352e.js | 528.00b | +0.00b | 710.00ms | +105.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 714.00ms | +96.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/271307026650366477)**

File: packages/core/core/src/BundleGraph.js [hunk 1]
```diff
@@ -812,16 +814,25 @@ export default class BundleGraph {
...
+          return node.value;
+        })
+        // Sort by bundleIds but reversed because the preorder DFS traversal in
+        // the bundler means that dependencies are added last (and need to run first)
+        .sort(
+          (a, b) =>
+            bundleGroup.bundleIds.indexOf(b.id) -
+            bundleGroup.bundleIds.indexOf(a.id),
+        )
+    );
```
Comment (devongovett) [inline packages/core/core/src/BundleGraph.js:835]: Would it be faster to do something like this?

```js
bundleGroup.bundleIds.map(id => nullthrows(this._graph.getNode(id)).value)
```

Comment (mischnic) [reply packages/core/core/src/BundleGraph.js]: Yeah, my worry was that `bundleGroup.bundleIds` and `getNodesConnectedFrom` could become out of sync (which actually was the case)

Comment (mischnic) [reply packages/core/core/src/BundleGraph.js]: Done in 3265bb7315075a3eb3fc61dc70804b5a82a21fa2, I've had to do some other changes to the bundle graph to keep `bundleIds` in sync with the graph

**Pull request merged**